docs/zero_to_hero_guide/00_Inference101.ipynb

```diff
@@ -0,0 +1,421 @@
+    "        conversation_history.append(assistant_message)\n",
+    "\n",
+    "# Use `await` in the Jupyter Notebook cell to call the function\n",
+    "await chat_loop()\n",
+    "# To run it in a python file, use this line instead\n",
+    "# asyncio.run(chat_loop())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "03fcf5e0",
+   "metadata": {},
+   "source": [
+    "## Streaming Responses\n",
+    "\n",
+    "Llama Stack offers a `stream` parameter in the `chat_completion` function, which allows partial responses to be returned progressively as they are generated. This can enhance user experience by providing immediate feedback without waiting for the entire response to be processed.\n",
+    "\n",
+    "### Example: Streaming Responses"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "d119026e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[32mUser> Write me a 3 sentence poem about llama\u001b[0m\n",
+      "\u001b[36mAssistant> \u001b[0m\u001b[33mSoft\u001b[0m\u001b[33mly\u001b[0m\u001b[33m padded\u001b[0m\u001b[33m feet\u001b[0m\u001b[33m on\u001b[0m\u001b[33m the\u001b[0m\u001b[33m ground\u001b[0m\u001b[33m,\n",
+      "\u001b[0m\u001b[33mA\u001b[0m\u001b[33m gentle\u001b[0m\u001b[33m llama\u001b[0m\u001b[33m's\u001b[0m\u001b[33m peaceful\u001b[0m\u001b[33m sound\u001b[0m\u001b[33m,\n",
+      "\u001b[0m\u001b[33mF\u001b[0m\u001b[33murry\u001b[0m\u001b[33m coat\u001b[0m\u001b[33m and\u001b[0m\u001b[33m calm\u001b[0m\u001b[33m,\u001b[0m\u001b[33m serene\u001b[0m\u001b[33m eyes\u001b[0m\u001b[33m all\u001b[0m\u001b[33m around\u001b[0m\u001b[33m.\u001b[0m\u001b[97m\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "import asyncio\n",
+    "from llama_stack_client import LlamaStackClient\n",
+    "from llama_stack_client.lib.inference.event_logger import EventLogger\n",
+    "from llama_stack_client.types import UserMessage\n",
+    "from termcolor import cprint\n",
+    "\n",
+    "async def run_main(stream: bool = True):\n",
+    "    client = LlamaStackClient(base_url=f'http://{HOST}:{PORT}')\n",
+    "\n",
+    "    message = UserMessage(\n",
+    "        content='Write me a 3 sentence poem about llama', role='user'\n",
+    "    )\n",
+    "    cprint(f'User> {message.content}', 'green')\n",
+    "\n",
+    "    response = client.inference.chat_completion(\n",
+    "        messages=[message],\n",
+    "        model='Llama3.2-11B-Vision-Instruct',\n",
+    "        stream=stream,\n",
+    "    )\n",
+    "\n",
+    "    if not stream:\n",
+    "        cprint(f'> Response: {response.completion_message.content}', 'cyan')\n",
+    "    else:\n",
+    "        async for log in EventLogger().log(response):\n",
+    "            log.print()\n",
+    "    \n",
+    "    models_response = client.models.list()\n",
```

Comment:
models_response was never used in this cell, should be deleted?

Reply:
removed